hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1130,6 +1142,24 @@ def example_wise_shrink(self, shrinker, **kwargs):
                 **kwargs
             )
 
+    @derived_value
+    def fine_grained_partitions(self):
```

Comment:
The main thing that helped me here was realizing that “partition” refers to a set of byte positions that mark the boundaries of all examples up to a particular example depth.

(Of course, the returned partitions are sorted unique lists instead of actual sets.)